flink-streaming-connectors/flink-connector-elasticsearch2/src/main/java/org/apache/flink/streaming/connectors/elasticsearch2/ElasticsearchSink.java

```diff
@@ -153,9 +165,90 @@ public ElasticsearchSink(Map<String, String> userConfig, List<InetSocketAddress>
 	 */
 	@Override
 	public void open(Configuration configuration) {
+		connect();
+
+		params = ParameterTool.fromMap(userConfig);
+
+		if (params.has(CONFIG_KEY_CONNECTION_RETRIES)) {
+			this.connectionRetries = params.getInt(CONFIG_KEY_CONNECTION_RETRIES);
+		}
+
+		buildBulkProcessorIndexer(client);
```

Comment:
Same here. Should try to avoid side-effecting class field members within utility member functions.